### DISCUSSION TITLE
如何制作自己 的组件库？

### DISCUSSION THREAD

**Discussion by xdqsgit** [ofajs/discussion#8] [Q&A] [answered]
如果我在一个项目中抽象了很多组件，如何方便的给下一个项目继续使用？
直接复制走？

**Comment by kirakiray** [comment, marked as answer]
有两个方案：
1. 放到公用的静态服务器空间；
2. 直接拷贝所有组件文件，放到下个项目中使用；

我推荐放到公用的静态服务器空间上，例如我这个 [Punch UI](https://github.com/ofajs/Punch-UI) 做了一大堆组件，就放到github的静态空间上，然后通过免费的[jsdelivr CDN](https://www.jsdelivr.com/)引用；例如我要使用这个仓库的 Punch UI 的按钮组件：

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>test button</title>
    <script src="https://cdn.jsdelivr.net/gh/kirakiray/ofa.js@4.4.5/dist/ofa.min.js"></script>
    <script src="https://cdn.jsdelivr.net/gh/ofajs/Punch-UI@0.1.8/init.js" type="module"></script>
  </head>
  <body>
    <l-m
      src="https://cdn.jsdelivr.net/gh/ofajs/Punch-UI@0.1.8/packages/button/button.html"
    ></l-m>

    <p-button> I am Test Button</p-button>
  </body>
</html>
```

通过 `<l-m>` 标签引用公共空间上的组件地址，就能直接使用了。

如果组件是私有的，可以放到公司提供（或自己购买）的公共空间或者付费cdn服务器上，同样适用 `<l-m>` 组件进行引用来使用相应的组件。

**Comment by xdqsgit** [comment]
明白，我在想是否有必要做一个转译器转把这些组件转成js，方便分发

**Reply by kirakiray** [reply]
是把一堆组件打包成一个js文件吗？有些项目为了减少请求数会需要。不过打包成一个的话，更新应用就变成全量更新文件了。
